files/fail2ban-abuseipdb.sh

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env bash
+#
+# Description:
+#   This script serves as a Fail2Ban `actionban` to report offending IP addresses to AbuseIPDB with custom COMMENTS.
+#   Mainly whenever Fail2Ban restarts, it calls the actionban function for each IP stored in the database file.
+#   If you restart your server often this script prevents duplicate reporting to AbuseIPDB by maintaining a local list of already reported IPs.
+#   Before reporting, it checks both the local list and AbuseIPDB to ensure the IP hasn't
+#   been reported previously, thereby avoiding redundant entries and potential API rate limiting.
+#
+# Integration with Fail2Ban:
+#   - Place this script in the 'action.d' directory of Fail2Ban.
+#   - Configure Fail2Ban to use this script as the `actionban`.
+#   - First edit 'abuseipdb.conf' in '/etc/fail2ban/action.d/abuseipdb.conf' and add following rule,
+#   - actionban = /etc/fail2ban/action.d/fail2ban-abuseipdb.sh \
+#         "<abuseipdb_apikey>" "<matches>" "<ip>" "<abuseipdb_category>" "<bantime>"
```

Comment:
Hmm... questionable thing to integrate this script with the action.
Especially because it is not advised to adjust stock configuration files (because of possible conflicts by update).
One can surely create `action.d/abuseipdb.local` instead and put it there, but...

Reply:
You're absolutely right, placing the configuration in `action.d/abuseipdb.local` is a better choice to keep the default untouched. 

I also wanted to clarify that the purpose of this PR is not to modify the core functionality of `abuseipdb.conf` directly. The script is intended as a isolated helper script, and it’s entirely up to you whether to keep it completely isolated in repo or link it to the `abuseipdb.conf` as the default configuration.

I’ve been using this script in a production environment for over a year with positive results, which is why I thought it might be helpful to share it with others through this PR.  https://www.abuseipdb.com/user/50466